File: maps-samplej/src/main/java/com/omh/android/maps/sample/start/MainActivity.java
package com.omh.android.maps.sample.start;

import static com.omh.android.maps.sample.utils.Constants.LAT_PARAM;
import static com.omh.android.maps.sample.utils.Constants.LNG_PARAM;

import android.content.Intent;
import android.os.Bundle;

import com.omh.android.maps.api.presentation.models.OmhCoordinate;
import com.omh.android.maps.sample.j.NavGraphDirections;
import com.omh.android.maps.sample.j.R;
import com.omh.android.maps.sample.j.databinding.ActivityMainBinding;
import com.omh.android.maps.sample.j.NavGraphDirections.ActionGlobalMapFragment;

import androidx.appcompat.app.AppCompatActivity;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration appBarConfiguration;
    private ActivityMainBinding binding;
    private NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setSupportActionBar(binding.toolbar);
        navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        appBarConfiguration = new AppBarConfiguration.Builder(navController.getGraph()).build();
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        handleIntent(getIntent());
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        handleIntent(intent);
    }

    private void handleIntent(Intent intent) {
        if (intent.getData() == null) return;
        String latitude = intent.getData().getQueryParameter(LAT_PARAM);
        String longitude = intent.getData().getQueryParameter(LNG_PARAM);
        if (latitude != null && longitude != null) {
            OmhCoordinate coordinate = new OmhCoordinate(Double.parseDouble(latitude), Double.parseDouble(longitude));
            ActionGlobalMapFragment action = NavGraphDirections.actionGlobalMapFragment();
            action.setCoordinate(coordinate);
            navController.navigate(action);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        return NavigationUI.navigateUp(navController, appBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: maps-sample/src/main/java/com/omh/android/maps/sample/start/MainActivity.kt
/*
 * Copyright 2023 Open Mobile Hub
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.omh.android.maps.sample.start

import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import com.omh.android.maps.api.presentation.models.OmhCoordinate
import com.omh.android.maps.sample.NavGraphDirections
import com.omh.android.maps.sample.R
import com.omh.android.maps.sample.databinding.ActivityMainBinding
import com.omh.android.maps.sample.utils.Constants.LAT_PARAM
import com.omh.android.maps.sample.utils.Constants.LNG_PARAM

class MainActivity : AppCompatActivity() {
    private lateinit var appBarConfiguration: AppBarConfiguration
    private val binding: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        handleIntent(intent)
        setSupportActionBar(binding.toolbar)
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        appBarConfiguration = AppBarConfiguration(navController.graph)
        setupActionBarWithNavController(navController, appBarConfiguration)
    }

    @SuppressLint("MissingSuperCall") // Android error: https://issuetracker.google.com/issues/67035929
    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        handleIntent(intent)
    }

    private fun handleIntent(intent: Intent?) {
        intent?.data?.let { uri: Uri ->
            val latitude = uri.getQueryParameter(LAT_PARAM)
            val longitude = uri.getQueryParameter(LNG_PARAM)
            if (latitude != null && longitude != null) {
                val coordinate = OmhCoordinate(latitude.toDouble(), longitude.toDouble())
                val action = NavGraphDirections.actionGlobalMapFragment(coordinate)
                findNavController(R.id.nav_host_fragment_content_main).navigate(action)
            }
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }
}
